providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -532,7 +559,21 @@ private <ValT extends Object, ReqT extends Message, ResT extends Message> Provid
                 .build();
 
         // run the referenced resolver method
-        ResT response = resolverRef.apply((ReqT) req);
+        final ResT response;
+
+        if (tracer != null) {
+            final Span span = tracer.spanBuilder("resolve")
+                    .setSpanKind(SpanKind.CLIENT)
+                    .startSpan();
+            span.setAttribute("feature_flag.key", key);
+            try (Scope scope = span.makeCurrent()) {
+                response = resolverRef.apply((ReqT) req);
+            } finally {
+                span.end();
```

Comment:
here I would inspect the response to add other attributes, ie ff.variant or exceptions, and set the span status in case of error.

Reply:
Right now we do not convey any evaluation error through the response. So this is not possible. See the protobuf definition of the contract [1] 

[1] - https://github.com/open-feature/schemas/blob/main/protobuf/schema/v1/schema.proto